products/memgraph/upcoming/tutorials/wiki-articles.md

```diff
@@ -0,0 +1,182 @@
+r.count as Count
+ORDER BY r.count DESC
+LIMIT 10;
+```
+
+
+6. Absolute values are fine, but how we get [term frequency](https://en.wikipedia.org/wiki/Tf%E2%80%93idf#Term_frequency_2) for our `Terms`? Notice we propagate `total_terms` with `WITH`.
+```opencypher
+MATCH (a:Article {name: "Protein"})-[r:CONTAINS]->(t:Term)
+WITH sum(r.count) AS total_terms
+
+MATCH (a:Article {name: "Protein"})-[r:CONTAINS]->(t:Term)
+WHERE toLower(a.name) != toLower(t.name)
+AND toLower(a.name+"s") != toLower(t.name)
+WITH r, t, total_terms
+
+RETURN DISTINCT
+t.name as Term,
+r.count as `Term count`,
+total_terms as `Total number of terms`,
+toFloat(r.count) / total_terms as term_frequency
+ORDER BY term_frequency
+DESC LIMIT 10;
+```
+
+7. Stopwords like "where", "or" and "make" usually show up in Wikipedia articles and other texts. Thankfully, this dataset doesn't contain any stopwords. But how can we, in general, know how important a `Term` is in an `Article` ? It may be the case that the `Term` appears in multiple `Articles` very often. Therefore, it's not so important and doesn't give us much new information. [tf-idf](https://en.wikipedia.org/wiki/Tf%E2%80%93idf) is a popular numerical statistic that gives us the answer to our problem. It takes into account how often does a `Term` appear in the `Article` "Protein" and how often in `Articles` overall. Let's calculate `tf_idf` for all `Terms` in an `Article` "Protein" 
+
+```opencypher
+MATCH (:Article)
+WITH count(*) AS number_of_articles
+
+MATCH (a:Article {name: "Protein"})-[r:CONTAINS]->(t:Term)
+WITH sum(r.count) AS terms_in_protein, number_of_articles
+
+MATCH (a:Article {name: "Protein"})-[r:CONTAINS]->(t:Term)<-[r2:CONTAINS]-(total_articles:Article)
+WHERE toLower(a.name) != toLower(total_articles.name)
+AND toLower(a.name) != toLower(t.name)
+AND toLower(a.name+"s") != toLower(t.name)
+WITH DISTINCT total_articles, t, toFloat(r.count) as term_in_protein, terms_in_protein, number_of_articles
+
+
+RETURN DISTINCT
+t.name as Term,
+term_in_protein / terms_in_protein as tf, 
+(-log(toFloat(count(total_articles)+1)/number_of_articles)) as idf,
+((term_in_protein / terms_in_protein) * (-log(toFloat(count(total_articles)+1)/number_of_articles))) as tf_idf
+
+ORDER BY tf_idf DESC
+LIMIT 10;
+```
+
+
+8. Let's find a peaceful `Articles` that contain a `Term` "Croatia"/"croatia" and doesn't contain the `Term` "war". `Articles` whose `name` contains "war" are also excluded.
+```opencypher
+MATCH (term1:Term)<-[r1:CONTAINS]-(a:Article)-[r2:CONTAINS]->(term2:Term)
+WHERE toLower(term1.name) != "war"
+AND NOT toLower(a.name) =~ ".*war.*"
+AND toLower(term2.name) = "croatia"
+RETURN DISTINCT a.name as `Article`, r2.count as ` "croatia" term count`
+ORDER BY r2.count DESC LIMIT 10;
+```
+
+
+9. Which `Categories` contain a substring of their child `Article's` names? Symbol `=~` denotes a [regular expression](https://docs.memgraph.com/memgraph/reference-overview/reading-existing-data#regular-expressions).
```

Comment:
Should this be `Article` and `Term's` or the query is wrong by not checking Category?